cpp/tests/quantiles/percentile_approx_test.cpp

```diff
@@ -127,94 +127,99 @@ void percentile_approx_test(cudf::column_view const& _keys,
-      std::vector<std::unique_ptr<cudf::groupby_aggregation>> aggregations;
-      aggregations.push_back(cudf::make_tdigest_aggregation<cudf::groupby_aggregation>(delta));
-      requests.push_back({values, std::move(aggregations)});
-      return std::move(gb.aggregate(requests).second[0].results[0]);
-    };
-    groupby_parts.push_back(cudf::type_dispatcher(values[idx].type(),
-                                                  percentile_approx_dispatch{},
-                                                  groupby,
-                                                  values[idx],
-                                                  delta,
-                                                  percentages,
-                                                  ulps));
-
-    // via reduce
-    auto reduce = [](cudf::column_view const& values, int delta) {
-      // result is a scalar, but we want to extract out the underlying column
-      auto scalar_result =
-        cudf::reduce(values,
-                     *cudf::make_tdigest_aggregation<cudf::reduce_aggregation>(delta),
-                     cudf::data_type{cudf::type_id::STRUCT});
-      auto tbl = static_cast<cudf::struct_scalar const*>(scalar_result.get())->view();
-      std::vector<std::unique_ptr<cudf::column>> cols;
-      std::transform(
-        tbl.begin(), tbl.end(), std::back_inserter(cols), [](cudf::column_view const& col) {
-          return std::make_unique<cudf::column>(col);
-        });
-      return cudf::make_structs_column(tbl.num_rows(), std::move(cols), 0, rmm::device_buffer());
-    };
-    // groupby path
-    reduce_parts.push_back(cudf::type_dispatcher(values[idx].type(),
-                                                 percentile_approx_dispatch{},
-                                                 reduce,
-                                                 values[idx],
-                                                 delta,
-                                                 percentages,
-                                                 ulps));
-  }
-
-  // second pass. run the percentile_approx with all the keys in one pass and make sure we get the
-  // same results as the concatenated by-key results.
-  std::vector<cudf::column_view> part_views;
-  std::transform(groupby_parts.begin(),
-                 groupby_parts.end(),
-                 std::back_inserter(part_views),
-                 [](std::unique_ptr<cudf::column> const& c) { return c->view(); });
-  auto expected = cudf::concatenate(part_views);
-
-  cudf::groupby::groupby gb(k);
-  std::vector<cudf::groupby::aggregation_request> requests;
-  std::vector<std::unique_ptr<cudf::groupby_aggregation>> aggregations;
-  aggregations.push_back(cudf::make_tdigest_aggregation<cudf::groupby_aggregation>(delta));
-  requests.push_back({_values, std::move(aggregations)});
-  auto gb_result = gb.aggregate(requests);
+  bool disable_cpu_cluster_computation[2] = {true, false};
+  for (int idx = 0; idx < 2; idx++) {
```

Comment:
nit: might be better not using the same loop variable same `idx` for this and the nested loop at L144.